src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -695,16 +727,9 @@ private async Task<bool> TrySendingBatchToClient(DocumentsOperationContext docsC
                         if (_buffer.Length > Constants.Size.Megabyte ||
                             sendingCurrentBatchStopwatch.ElapsedMilliseconds > 1000)
                         {
-                            if (docsToFlush > 0)
```

Comment:
If there is nothing to flush, nothing will be sent the user.
Why did you remove the `SendHeartBeat` ?

Reply:
There is no way docsToFlush will be 0, look few lines above